ts/keys.go

```diff
@@ -78,31 +78,33 @@ func MakeDataKey(name string, source string, r Resolution, timestamp int64) prot
 }
 
 // DecodeDataKey decodes a time series key into its components.
-func DecodeDataKey(key proto.Key) (string, string, Resolution, int64) {
-	var (
-		name          []byte
-		source        []byte
-		resolutionInt int64
-		timeslot      int64
-		remainder     = key
-	)
-
+func DecodeDataKey(key proto.Key) (string, string, Resolution, int64, error) {
```

Comment:
FWIW, I don't think the returned `error` is a good reason to use named returns. But the previous returned values are (i.e. what is the difference between the 2 strings).